### PR TITLE
[OP] Softmax

### PR DESCRIPTION
**Pull request by anadeem2**
## Description ##
Softmax OP lowering. For some reason, the tests only pass when dim = last dim (-1). This can be implicitly inferred, but providing wrong dim causes test to fail (that is why pytest.mark is not provided). I checked in RAF and the axis is also set to -1. 

## Checklist ##

- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

cc @awslabs/raf-reviewer

### CODE REVIEW

**Comment by anadeem2** [comment]
> Is the dim correctly passed? For example, if you set 0 in pytorch, will RAF get 0 correctly?
In ```/third_party/raf/src/op/declare/nn.cc``` for softmax, I tried std::cout in RAF to print the ndim, I also tried to put false ltc_checks to crash and print softmax dim, but despite all this I cannot get a verification from RAF. Do you have a
We can close this PR, as softmax is now a part of #41  and move discussion there.
